### ISSUE TITLE
ArgumentError: bad argument (expected URI object or URI string)

### ISSUE THREAD

**Issue by slayer** [rubysec/bundler-audit#235]
bundler-audit starts failing today at CI: 

```
$ cd /app && bundle exec bundle-audit update && bundle exec bundle-audit
Updating ruby-advisory-db ...
Cloning into '/home/app/.local/share/ruby-advisory-db'...
Updated ruby-advisory-db
ruby-advisory-db: 422 advisories
bundler: failed to load command: bundle-audit (/app/vendor/bundle/ruby/2.6.0/bin/bundle-audit)
ArgumentError: bad argument (expected URI object or URI string)
  /usr/local/lib/ruby/2.6.0/uri/common.rb:739:in `URI'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:170:in `internal_source?'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:113:in `block (2 levels) in scan_sources'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:112:in `each'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:112:in `block in scan_sources'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:102:in `map'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:102:in `scan_sources'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/scanner.rb:74:in `scan'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/lib/bundler/audit/cli.rb:44:in `check'
  /app/vendor/bundle/ruby/2.6.0/gems/thor-0.19.4/lib/thor/command.rb:27:in `run'
  /app/vendor/bundle/ruby/2.6.0/gems/thor-0.19.4/lib/thor/invocation.rb:126:in `invoke_command'
  /app/vendor/bundle/ruby/2.6.0/gems/thor-0.19.4/lib/thor.rb:369:in `dispatch'
  /app/vendor/bundle/ruby/2.6.0/gems/thor-0.19.4/lib/thor/base.rb:444:in `start'
  /app/vendor/bundle/ruby/2.6.0/gems/bundler-audit-0.6.1/bin/bundle-audit:10:in `<top (required)>'
  /app/vendor/bundle/ruby/2.6.0/bin/bundle-audit:23:in `load'
  /app/vendor/bundle/ruby/2.6.0/bin/bundle-audit:23:in `<top (required)>'
ERROR: Job failed: exit code 1
```

**Comment by dmtask** [comment]
I think i have the same Problem. My Gitlab Pipelines failed, because the Command `bundle exec bundle-audit` give a Error `Could not find command "audit".`

**Comment by olegantonyan** [comment]
The problem is that we got `Bundler::URI::HTTP` instead of `URI::HTTP`. This comes from bundler itself from `Bundler::LockfileParser` class.
Downgrading bundler from 2.1.2 to 2.0.2 resolves the issue.
Reported in bundler https://github.com/bundler/bundler/issues/7511

**Comment by dmtask** [comment]
> Downgrading bundler from 2.1.2 to 2.0.2 resolves the issue.

Yes, this was my solution. :)

**Comment by ThomasKoppensteiner** [comment]
@olegantonyan, @slayer and @dmtask do you think my pull request is helpful?

**Comment by dmtask** [comment]
@ThomasKoppensteiner Maybe, but I'm not sure. Is the fix working on your tests? (I can't test it right now. I am currently running version 2.0.2 stable).

**Comment by flaprano** [comment]
I have this same problem.  I found out yesterday that it only happened to my internal rubygems and also because it uses HTTP.  If the address uses HTTPS, it is not attempted to cast the URI object

**Comment by chrislewis60** [comment]
Yeah, make sure you reference the rubygems **https** address in your Gemfile, and re-run bundle if you update it from http. I was updating an old project that referenced http when I encountered this issue. 
`source 'https://rubygems.org'`

**Comment by ThomasKoppensteiner** [comment]
FYI: I closed my PR (#240) because a similar fix (#236) has already been merged.

**Comment by ThomasKoppensteiner** [comment]
Can someone from @rubysec (@indirect, @brynary or @hsbt) please release the already merged fix from #236?

**Comment by indirect** [comment]
@ThomasKoppensteiner I'm not sure why you think any of us have the ability to do that. You can easily see from https://rubygems.org/gems/bundler-audit that only @postmodern has permissions to release the gem.

**Comment by ThomasKoppensteiner** [comment]
Sorry for bothering you.
I only looked at the member list here at Github and didn't checked the owner at rubygems.

**Comment by postmodern** [comment]
This was [finally] released in 0.7.0.
